src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -120,12 +130,50 @@ private String formatTableList(String caption, List<String> items) {
     sb.append("|");
     sb.append(NEW_LINE);
     for (String item : items) {
-      sb.append("| ").append(item).append(" |").append(NEW_LINE);
+        sb.append("| ").append(item).append(" |").append(NEW_LINE);
+        if (this.includeFilesInOverview > 0 && issueToFilesMap.containsKey(item)) {
+            String files = String.join(", ", issueToFilesMap.get(item));
+            sb.append("| :arrow_right_hook: *Files: ").append(files).append("* |").append(NEW_LINE);
+        }
     }
 
     return sb.toString();
   }
 
+  private ListMultimap<String, String> getMappedIssuesToFiles(Collection<PostJobIssue> allIssues) {
+    ListMultimap<String, String> issueToFilesMap = ArrayListMultimap.create();
+    if (this.includeFilesInOverview <= 0) {
+      return issueToFilesMap;
+    }
+
+    for (PostJobIssue issue : allIssues) {
+      String title = this.printIssueMarkdown(issue);
+
+      Integer issueLine = issue.line();
+      if (issueLine == null) {
+        issueLine = 0;
+      }
+
+      issueToFilesMap.get(title).add(this.issuePathResolver.getIssuePath(issue) + ":" + issueLine);
+    }
+
+    this.sortMappedIssuesToFiles(issueToFilesMap);
+
+    return issueToFilesMap;
+  }
+
+  private void sortMappedIssuesToFiles(ListMultimap<String, String> issueToFilesMap) {
+    for (String key : Lists.newArrayList(issueToFilesMap.keySet())) {

```

Comment:
I think it's unnecessary to wrap the keySet in a list.

Reply:
This is there to avoid getting `java.util.ConcurrentModificationException`

Reply:
Ok. Personally I would prefer to stick with a Set then, maybe using `Set.of()`, but honestly it doesn't matter.